subprojects/core/src/main/java/org/gradle/plugin/software/internal/SoftwareTypeImplementation.java

```diff
@@ -26,5 +28,7 @@ public interface SoftwareTypeImplementation {
 
     Class<?> getModelPublicType();
 
-    Class<? extends Plugin<?>> getPluginClass();
+    Class<? extends Plugin<Project>> getPluginClass();
```

Comment:
Maybe this could be called the `definingPluginClass`?

Reply:
I don't think we should do this, as having both a `getRegisteringPluginClass()` and a `getDefiningPluginClass()` in the interface will be confusing.

I think the central concept here is the `Plugin` that defines a `SoftwareType`, so we can call that just `pluginClass`.  Then we distinguish the Settings plugin that registers that plugin.  I also thought about calling this just `registeringClass` and dropping the Plugin from that entirely, but I thought it was better leave that in the name.